lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlobFileSystem.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureBlob
+{
+    /// <summary>
+    /// An IFileSystem implementation for Azure Blob storage.
+    /// A few calls are quite costly due to the inability to distinguish what type
+    /// of blob we are trying to access, so GetBlobReferenceFromServer is invoked.
+    /// There is currently no GetBlobReference call that does not populate
+    /// the fields of an ICloudBlob.
+    /// </summary>
+    public sealed class AzureBlobFileSystem : IFileSystem
+    {
+        private readonly ICloudBlobClient _client;
+
+        public AzureBlobFileSystem(Uri baseUri)
```

Comment:
Do we want to expose all those non-injectable constructors? 
You already have all the overloading constructors for AzureCloudBlobClient, what you need is an instance of AzureCloudBlobClient, you don't need to repeat the overloading constructors here. 
Also IFileSystem most possibly will be injected by client through Tang configuration. It is inconsistent to have non-injectable constructor here. If client choose different way to config IFileSystem, that amy cause confusion.